apps/web/src/ee/clerk/providers/useCreateAuthContextEnterprise.ts

```diff
@@ -0,0 +1,205 @@
+  name: clerkOrganization.name,
+  createdAt: clerkOrganization.createdAt.toString(),
+  updatedAt: clerkOrganization.updatedAt.toString(),
+  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
+  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
+  domain: clerkOrganization.publicMetadata.domain,
+  productUseCases: clerkOrganization.publicMetadata.productUseCases,
+});
+
+export function useCreateAuthContextEnterprise() {
+  const { signOut, orgId } = useAuth();
+  const { user: clerkUser, isLoaded: isUserLoaded } = useUser();
+  const { organization: clerkOrganization, isLoaded: isOrganizationLoaded } = useOrganization();
+  const { setActive, isLoaded: isOrgListLoaded } = useOrganizationList();
+
+  const [user, setUser] = useState<IUserEntity | undefined>(undefined);
+  const [organization, setOrganization] = useState<IOrganizationEntity | undefined>(undefined);
+
+  const ldClient = useLDClient();
+  const segment = useSegment();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const inPublicRoute = Array.from(PUBLIC_ROUTES_PREFIXES.values()).find((prefix) =>
+    location.pathname.startsWith(prefix)
+  );
+  const inPrivateRoute = !inPublicRoute;
+
+  const logout = useCallback(() => {
+    queryClient.clear();
+    segment.reset();
+    navigate(ROUTES.AUTH_LOGIN);
+    signOut();
+  }, [navigate, queryClient, segment, signOut]);
+
+  const redirectTo = useCallback(
+    ({
+      url,
+      redirectURL,
+      origin,
+      anonymousId,
+    }: {
+      url: string;
+      redirectURL?: string;
+      origin?: string;
+      anonymousId?: string | null;
+    }) => {
+      const finalURL = new URL(url, window.location.origin);
+
+      if (redirectURL) {
+        finalURL.searchParams.append('redirect_url', redirectURL);
+      }
+
+      if (origin) {
+        finalURL.searchParams.append('origin', origin);
+      }
+
+      if (anonymousId) {
+        finalURL.searchParams.append('anonymous_id', anonymousId);
+      }
+
+      // Note: Do not use react-router-dom. The version we have doesn't do instant cross origin redirects.
+      window.location.replace(finalURL.href);
+    },
+    []
+  );
+
+  const redirectToLogin = useCallback(
+    ({ redirectURL }: { redirectURL?: string } = {}) => redirectTo({ url: ROUTES.AUTH_LOGIN, redirectURL }),
+    [redirectTo]
+  );
+
+  const redirectToSignUp = useCallback(
+    ({
+      redirectURL,
+      origin,
+      anonymousId,
+    }: { redirectURL?: string; origin?: string; anonymousId?: string | null } = {}) =>
+      redirectTo({ url: ROUTES.AUTH_SIGNUP, redirectURL, origin, anonymousId }),
+    [redirectTo]
+  );
+
+  const switchOrgCallback = useCallback(async () => {
+    await queryClient.refetchQueries();
+  }, [queryClient]);
+
+  // check if user has active organization
+  useEffect(() => {
+    if (orgId) {
+      return;
+    }
+
+    if (isOrgListLoaded && clerkUser) {
```

Comment:
I am not so sure why we need this callback and this logic. All the org and user logic should be handled by Clerk components.

Reply:
It is a guard that checks if user has an active organization. Maybe I'm missing something but by default when I sign up or sign in as a user, the Clerk JWT doesn't contain orgId. 

Without this check you would be able to access other private routes without an active organization / creating the organization first. So we need to eitehr force creation of an org or if user is already a member of some orgs, we chose the first one as default - alternatively we could display org select after login but that can be done now in the org switch selector inside an app as well.

I mostly followed this guide: https://clerk.com/docs/guides/force-organizations